lib/api/CDataFrameAnalysisInstrumentation.cc

```diff
@@ -64,14 +69,14 @@ void CDataFrameAnalysisInstrumentation::writer(core::CRapidJsonConcurrentLineWri
     m_Writer = writer;
 }
 
-void CDataFrameAnalysisInstrumentation::nextStep(std::uint32_t /*step*/) {
-    // TODO reactivate state writing, once the Java backend can accept it
-    //    this->writeState(step);
+void CDataFrameAnalysisInstrumentation::nextStep(std::uint32_t step) {
+    this->writeState(step);
 }
 
 void CDataFrameAnalysisInstrumentation::writeState(std::uint32_t step) {
-    this->writeProgress(step);
-    this->writeMemory(step);
+    // this->writeProgress(step);
+    int64_t timestamp = core::CTimeUtils::toEpochMs(core::CTimeUtils::now());
+    this->writeMemory(timestamp);
```

Comment:
nit: use std::int64_t, or how about just `this->writeMemory(core::CTimeUtils::toEpochMs(core::CTimeUtils::now());`

Reply:
I will have to write the time stamp into analysis object as well. Thus, I prefer to keep it as a variable.